mobile/mobile-core/devices-service.ts

```diff
@@ -344,9 +344,20 @@ export class DevicesService extends EventEmitter implements Mobile.IDevicesServi
 	}
 
 	@exported("devicesService")
-	public deployOnDevices(deviceIdentifiers: string[], packageFile: string, packageName: string): Promise<void>[] {
+	public deployOnDevices(deviceIdentifiers: string[], packageFile: string, packageName: string | Mobile.IProjectIdentifier): Promise<void>[] {
 		this.$logger.trace(`Called deployOnDevices for identifiers ${deviceIdentifiers} for packageFile: ${packageFile}. packageName is ${packageName}.`);
-		return _.map(deviceIdentifiers, deviceIdentifier => this.deployOnDevice(deviceIdentifier, packageFile, packageName));
+		return _.map(deviceIdentifiers, async deviceIdentifier => {
```

Comment:
you do not need the `async` keyword here

Reply:
I put it there because the method returns an array of promises and if getDeviceByIdentifier fails an exception was thrown instead of returning a rejected promise.